config/config.go

```diff
@@ -47,6 +47,7 @@ type SourceSubnet struct {
 	WSEndpoint            string                           `mapstructure:"ws-endpoint" json:"ws-endpoint"`
 	MessageContracts      map[string]MessageProtocolConfig `mapstructure:"message-contracts" json:"message-contracts"`
 	SupportedDestinations []string                         `mapstructure:"supported-destinations" json:"supported-destinations"`
+	CatchUpBlockHeight    uint64                           `mapstructure:"catch-up-block-height" json:"catch-up-block-height"`
```

Comment:
I'm a bit worried we're providing too many options around the start block that the relayer will use on each initialization.

For instance, I think this is a weird possible case:
1. New relayer is started with no catch up block height and no DB. Uses current block and writes it to disk.
2. Now user realizes they wanted to relay a transaction from before starting it, so they specify a catch up block height and restart. 

In this case, the catch up block height has no effect because it was run without a DB or initial setting. 

I'd be in favor of simplifying the flow by removing the final case of "if there is no stored or configured start block, start from the tip of the chain", and renaming `CatchUpBlockHeight` to `StartBlockHeight`. If neither is present, we could print a warning log saying that the relayer will relay from the start of the chain, and to set the `StartBlockHeight` value if this is not the desired behavior.

Reply:
Is your suggestion to have `StartBlockHeight` potentially re-process blocks if the provided value is lower than the height stored in the DB?

I think that's reasonable, since we check that the message isn't already marked as delivered in the Teleporter contract before sending a tx.

Reply:
Discussed offline, and the approach we're going to go with is:
- When intializing a relayer for the first time (empty DB), if `StartBlockHeight` is provided, start from there, otherwise start from block 0 and emit a warning
- Otherwise, choose between the max of the `StartBlockHeight` and the block height in the DB

If the user wishes to start from a specific block earlier than the one stored in the DB, then the workaround would be to delete that entry from the DB and provide the desired block as the `StartBlockHeight` argument.

In the future, we'll add a config option to specify Warp message IDs to relay, which will provide the user fine-grained control for catching up on missed messages that were not automatically processed, without having to manually modify the DB. Issue here: https://github.com/ava-labs/awm-relayer/issues/126